src/formatDate.js

```diff
@@ -8,7 +8,50 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  let year;
+  let month;
+  let day;
+  const formattedDate = [];
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i] === 'YYYY') {
+      year = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'YY') {
+      year = dateParts[i] < 30 ? `20${dateParts[i]}` : `19${dateParts[i]}`;
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = dateParts[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = dateParts[i];
+    }
+  }
+
+  for (const format of toFormat) {
+    switch (format) {
+      case 'YYYY':
+        formattedDate.push(year);
+        break;
+
+      case 'YY':
+        formattedDate.push(year.slice(-2));
+        break;
+
+      case 'MM':
+        formattedDate.push(month);
+        break;
+
+      case 'DD':
+        formattedDate.push(day);
+    }
+  }
+
+  return formattedDate.join(toFormat[3]);
```

Comment:
The implementation of the `formatDate` function seems correct, but there are a few edge cases that are not handled. For example, if the `fromFormat` or `toFormat` arrays have elements other than 'YYYY', 'YY', 'MM', 'DD', or the separator, the function will not behave correctly. It would be beneficial to add validation to ensure that the input formats only contain valid format specifiers. Additionally, consider edge cases where the input date string does not match the expected `fromFormat` structure, which could lead to incorrect parsing and formatting.